src/main/java/com/hackclub/hccore/advancements/MileAdv.java

```diff
@@ -26,9 +26,13 @@ public MileAdv(Advancement root, AdvancementKey key,
     super(key.getKey(), displayBuilder.coords(adapter, key).build(), root, maxProgression);
 
     registerEvent(PlayerMoveEvent.class, e -> {
-      if (e.getPlayer().getStatistic(Statistic.WALK_ONE_CM) >= 1609.344 * 100 * 1000) {
-        grant(e.getPlayer());
-      }
+        if (isGranted(e.getPlayer())) {
+            return;
+        }
+        
+        if (e.getPlayer().getStatistic(Statistic.WALK_ONE_CM) >= 160934400) {
```

Comment:
Personally, the magic number being (slightly) split out made a bit of sense
1000 miles to Kms
Centimetres per meter
Meters per kilometre